source/frontend/src/features/leases/add/AddLeaseContainer.tsx

```diff
@@ -1,97 +1,106 @@
+import { ReactComponent as Fence } from 'assets/images/fence.svg';
 import GenericModal from 'components/common/GenericModal';
-import { SidebarStateContext } from 'components/layout/SideNavBar/SideNavbarContext';
-import { SidebarContextType } from 'components/layout/SideNavBar/SideTray';
+import { useMapSearch } from 'components/maps/hooks/useMapSearch';
 import { MapStateActionTypes, MapStateContext } from 'components/maps/providers/MapStateContext';
-import { AddLeaseLayout, LeaseBreadCrumb, LeaseHeader, LeaseIndex } from 'features/leases';
+import MapSideBarLayout from 'features/mapSideBar/layout/MapSideBarLayout';
+import SidebarFooter from 'features/properties/map/shared/SidebarFooter';
 import { FormikProps } from 'formik';
 import { Api_Lease } from 'models/api/Lease';
 import * as React from 'react';
-import { useContext, useRef, useState } from 'react';
-import { useHistory } from 'react-router';
+import { useMemo, useState } from 'react';
+import { useEffect, useRef } from 'react';
+import { useHistory } from 'react-router-dom';
 
-import { LeasePageNames, leasePages } from '../detail/LeaseContainer';
 import { useAddLease } from '../hooks/useAddLease';
-import { FormLease } from '../models';
+import { LeaseFormModel } from '../models';
 import AddLeaseForm from './AddLeaseForm';
 
-export interface IAddLeaseContainerProps {}
+export interface IAddLeaseContainerProps {
+  onClose: () => void;
+}
 
 export const AddLeaseContainer: React.FunctionComponent<
   React.PropsWithChildren<IAddLeaseContainerProps>
 > = props => {
-  const leasePage = leasePages.get(LeasePageNames.DETAILS);
-  const { setTrayPage } = useContext(SidebarStateContext);
-  const { setState } = useContext(MapStateContext);
-  const onClickManagement = () => setTrayPage(SidebarContextType.LEASE);
   const history = useHistory();
+  const formikRef = useRef<FormikProps<LeaseFormModel>>(null);
+  const { setState } = React.useContext(MapStateContext);
+  const initialForm = useMemo<LeaseFormModel>(() => {
+    return new LeaseFormModel();
```

Comment:
You may get a bunch of formik errors for this, remember that formik expects an object with all fields already defined (ie: '' instead of undefined).

Reply:
good point. Will check that the strings are actually defined